k9mail-library/src/test/java/com/fsck/k9/mail/filter/LineWrapOutputStreamTest.java

```diff
@@ -0,0 +1,233 @@
+            "comprehensive and cover every single feature, or even every commonly used feature. " +
+            "Instead, it introduces many of Python’s most noteworthy features, and will give you " +
+            "a good idea of the language’s flavor and style. After reading it, you will be able " +
+            "to read and write Python modules and programs, and you will be ready to learn more " +
+            "about the various Python library modules described in The Python Standard Library.  " +
+            " The Glossary is also worth going through.";
+
+    @Test
+    public void testWithForTrailingSpacePreservation() throws IOException {
+        Buffer baos = new Buffer();
+        writeToLineWrapStream(20, baos);
+        assertEquals("Mismatch with wrap length of 20", EXPECTED_WRAPPED_AT20, baos.readUtf8());
+        writeToLineWrapStream(40, baos);
+        assertEquals("Mismatch with wrap length of 40", EXPECTED_WRAPPED_AT40, baos.readUtf8());
+        writeToLineWrapStream(60, baos);
+        assertEquals("Mismatch with wrap length of 60", EXPECTED_WRAPPED_AT60, baos.readUtf8());
+        writeToLineWrapStream(80, baos);
+        assertEquals("Mismatch with wrap length of 80", EXPECTED_WRAPPED_AT80, baos.readUtf8());
+    }
+
+    private static void writeToLineWrapStream(int MAX_BYTES_PER_LINE, Buffer baos)
+            throws IOException {
+        baos.clear();
+        LineWrapOutputStream lwops = new LineWrapOutputStream(baos.outputStream(), MAX_BYTES_PER_LINE);
+        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(lwops), 100);
+        writer.write(INPUT_STRING);
+        writer.flush();
+    }
+
+    /**
+     * Feeds in some predefined texts without CRLF characters since we will be
+     * assessing optimal, non-overflowing, wrapping: empty, excessive, breaks
+     * would interfere.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testThatWrappingIsTight() throws Exception {
+        Buffer baos = new Buffer();
+        testTightWrapLongStatic(20, baos);
+        testTightWrapLongStatic(40, baos);
+        testTightWrapLongStatic(60, baos);
+        testTightWrapLongStatic(80, baos);
+        testTightWrapLongStatic(100, baos);
+        testTightWrapLongStatic(120, baos);
+    }
+
+    private static void testTightWrapLongStatic(final int MAX_BYTES_PER_LINE
+            , Buffer baos) throws IOException {
+        baos.clear();
+        LineWrapOutputStream lwops = new LineWrapOutputStream(baos.outputStream(), MAX_BYTES_PER_LINE);
+        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(lwops), 100);
+        fillWithStaticData(writer);
+        checkWrapsTight(MAX_BYTES_PER_LINE, baos.readByteString());
+    }
+
+    private static void fillWithStaticData(BufferedWriter writer) throws IOException {
+        writer.write(someLongString);
+        writer.flush();
+    }
+
+    private static void checkWrapsTight(int MAX_BYTES_PER_LINE, ByteString baos) {
```

Comment:
This is way too much logic for tests. We have static input, so we can hardcode the expected output.